build.rs

```diff
@@ -230,13 +230,29 @@ fn parse_header_defines(header_path: impl AsRef<Path>) -> Result<HashMap<String,
     Ok(definitions)
 }
 
-fn fix_config_map(mut config_map: HashMap<String, String>) -> HashMap<String, String> {
+fn fix_config_map(
+    mut config_map: HashMap<String, String>,
+    interpreter_config: &InterpreterConfig,
+) -> HashMap<String, String> {
+    if interpreter_config.version.major == 3 && interpreter_config.version.minor.unwrap_or(0) > 7 {
```

Comment:
I suggest the following:

```suggestion
    assert_eq!(interpreter_config.version.major, 3);
    if interpreter_config.version.minor.unwrap_or(0) >= 8 {
```

Because we only support Python 3, I think it's good to just make this code fail if we ever try to run it on Python 4.

Reply:
Yeah, but is this a proper place to do this assertion....?
I'd rather pass only the mimor version to this function.

Reply:
Passing just the minor version instead of the full `interpreter_config` to this function also seems like a good idea!

Reply:
will we support python 4? if support, we should keep the interpreter_config as the function argument. and leave room for the possible change in the python 4.